example-github/src/main/java/feign/example/github/GitHubExample.java

```diff
@@ -53,4 +64,51 @@ public static void main(String... args) throws InterruptedException {
     String login;
     int contributions;
   }
+
+  static class ClientError {
```

Comment:
noto sure about gson. with jackson it would be just about annotating the constructor. but the JacksonDecoder needs some configuration as it won't use non-public fields by default.